packages/medusa/src/api/routes/admin/products/update-variant.ts

```diff
@@ -106,20 +110,53 @@ import { validator } from "../../../../utils/validator"
 export default async (req, res) => {
   const { id, variant_id } = req.params
 
+  const manager: EntityManager = req.scope.resolve("manager")
+  const productService: ProductService = req.scope.resolve("productService")
+  const pricingService: PricingService = req.scope.resolve("pricingService")
+  const featureFlagRouter: FlagRouter = req.scope.resolve("featureFlagRouter")
   const validated = await validator(
     AdminPostProductsProductVariantsVariantReq,
     req.body
   )
 
   const validatedQueryParams = await validator(PriceSelectionParams, req.query)
 
-  const productService: ProductService = req.scope.resolve("productService")
-  const pricingService: PricingService = req.scope.resolve("pricingService")
+  if (featureFlagRouter.isFeatureEnabled(IsolatePricingDomainFeatureFlag.key)) {
+    const updateVariantsWorkflow = UpdateProductVariants.updateProductVariants(
+      req.scope
+    )
+
+    const input = {
+      productVariants: [
+        {
+          id: variant_id,
+          ...validated,
+        },
+      ] as WorkflowTypes.ProductWorkflow.UpdateProductVariantsInputDTO[],
+    }
+
+    await updateVariantsWorkflow.run({
+      input,
+      context: {
+        manager,
+      },
+    })
+
+    const rawProduct = await productService.retrieve(id, {
```

Comment:
question: Should we use the remote query here?

Reply:
suggestion: Eventually we should do something like this. 

Since this will depend on if the product isolation is enabled, I think we should not return early, rather we should adopt a similar pattern to the other endpoints where we run either the remote-query or the fetch after the update to keep the handler as simple as possible: 

```
export default async (req, res) => {
  if (featureFlagRouter.isFeatureEnabled(IsolatePricingDomainFeatureFlag.key)) {
    ...

    await updateVariantsWorkflow.run({
      input,
      context: {
        manager,
      },
    })
  } else {  
      await manager.transaction(async (transactionManager) => {
        await productVariantService
        .withTransaction(transactionManager)
        .update(variant_id, {
          product_id: id,
          ...validated,
        })
      })
  }

  let rawProduct

  if(featureFlagRouter.isFeatureEnabled(IsolateProductDomainFeatureFlag.key)) { 
    
    rawProduct = // Remote query result

  } else { 
    rawProduct = await productService.retrieve(id, {
      select: defaultAdminProductFields,
      relations: defaultAdminProductRelations,
      ...validatedQueryParams,
    })
  }

  const [product] = await pricingService.setProductPrices([rawProduct])

  res.json({ product })
}
```